app/models/post.rb

```diff
@@ -10,4 +10,7 @@ class Post < ApplicationRecord
   scope :sort_by_time, ->{order created_at: :desc}
   scope :lastest, ->{order created_at: :desc}
   scope :by_top, ->(topic_id){where topic_id: topic_id if topic_id.present?}
+
+  scope :search_by, ->(search_term){
+    where "LOWER(title) LIKE :search_term OR LOWER(body) LIKE :search_term", search_term: "%#{search_term.downcase}%"}
```

Comment:
```suggestion
    scope :search_by, (lambda do |params|
      where("LOWER(title) LIKE :key OR LOWER(body) LIKE :key", key: "%#{params}%")
    end)
```
You can try to use lambda